src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -806,28 +828,220 @@ public async Task Delete()
+
+                    foreach (var removedNodeTag in details.NodesToRemove) // nodeTags
+                    {
+                        var removeNodeFromDbCmdGuid = $"RemoveNodeFromDatabase/{removedNodeTag}/{deleteCmdIndex}";
+                        var removeNodeFromDbCmdRaftIndex = 0L;
+                        var sw = Stopwatch.StartNew();
+                        while (TryGetLastCommittedDeleteRaftIndexByGuid(ServerStore, removeNodeFromDbCmdGuid, out removeNodeFromDbCmdRaftIndex) == false)
+                        {
+                            if (sw.Elapsed > timeout)
+                            {
+                                throw new TimeoutException(
+                                    $"RemoveNodeFromDatabase command of node {removedNodeTag} doesn't exist in the log of node {ServerStore.NodeTag}.");
+                            }
+
+                            await Task.Delay(TimeSpan.FromMilliseconds(300));
+                        }
+
+                        using var cts = new CancellationTokenSource(timeout - sw.Elapsed);
+                        await WaitForDatabaseNotificationOnCluster(dbName, removeNodeFromDbCmdRaftIndex, nodesTagsAndUrls, databaseNodes, reqExecutors, cts.Token);
+                    }
+                }
+            }
+            finally
+            {
+                foreach (var requestExecutor in reqExecutors.Values)
+                    requestExecutor.Dispose();
+            }
+        }
+
+
+        private async Task WaitForDatabaseNotificationOnCluster(string databaseName, long index, 
+            Dictionary<string, string> clusterNodes, // (nodeTag, url)
+            HashSet<string> databaseNodes, // nodeTags
+            Dictionary<string, ClusterRequestExecutor> reqExecutors,
+            CancellationToken token)
+        {
+
+            foreach (var (tag, nodeUrl) in clusterNodes)
+            {
+                try
+                {
+                    ClusterRequestExecutor requestExecutor;
+                    if (reqExecutors.TryGetValue(tag, out requestExecutor) == false)
+                    {
+                        requestExecutor = reqExecutors[tag] =
+                            ClusterRequestExecutor.CreateForSingleNode(nodeUrl, Server.Certificate.Certificate, DocumentConventions.DefaultForServer);
+                    }
+
+                    using (requestExecutor.ContextPool.AllocateOperationContext(out var context))
+                    {
+                        if (databaseNodes.Contains(tag)) // if node is part of db topology - wait for commit and database notification
+                        {
+                            var cmd = new WaitForDatabaseIndexNotificationCommand(index, databaseName);
+                            await requestExecutor.ExecuteAsync(cmd, context, token: token);
+                        }
+                        else // else - wait for commit only
+                        {
```

Comment:
why?

Reply:
You're right, there is no point in doing that..
I removed it